code/game/objects/structures/shower.dm

```diff
@@ -253,17 +253,19 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/shower, (-16))
 	take_his_status_effect.remove_status_effect(/datum/status_effect/shower_regen)
 
 /obj/machinery/shower/proc/wash_atom(atom/target)
-	target.wash(CLEAN_RAD | CLEAN_WASH)
+	var/datum/reagent/blood/bloody_shower = locate(/datum/reagent/blood) in reagents.reagent_list
```

Comment:
I think you should be using `has_reagent` or `get_reagent` instead of `locate`

Reply:
Done as well. I took Annie's suggestions and calculated if the reagents are +70% determine if the shower is bloody or clean. Also if radioactive reagents are +20% the shower will not remove radiation. (so it's possible to have a sneaky radioactive/clean shower)